arkane/isodesmic.py

```diff
@@ -141,5 +143,102 @@ def calculate_target_thermo(self):
         return ScalarQuantity(target_thermo, 'J/mol')
 
 
+class SpeciesConstraints(object):
+    """
+    A class for defining and enumerating constraints to ReferenceSpecies objects for error canceling reactions
+    """
+
+    def __init__(self, target, reference_species, conserve_bonds=True, conserve_ring_size=True):
+        """
```

Comment:
ReferenceDatabase has a method `extract_model_chemistry` which return a list of `ErrorCancelingSpecies` with desired model_chemistry.  This list can then be passed to `SpeciesConstraints` as `reference_species` argument.  I think it might be also useful to have a method in the `ReferenceSpecies` class that converts `ReferenceSpecies` object to `ErrorCancelingSpecies` object.

Reply:
Thanks for the suggestion, this is actually a much better way to do it. I have also renamed the `reference_species` argument to `reference_list` to avoid confusion